core/src/main/java/com/airbnb/aerosolve/core/function/FunctionUtil.java

```diff
@@ -0,0 +1,77 @@
+package com.airbnb.aerosolve.core.function;
+
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.util.Arrays;
+
+public class FunctionUtil {
+  public static float[] fitPolynomial(float[] data) {
+    int numCoeff = 6;
+    int iterations = numCoeff * 4;
+    float[] initial = new float[numCoeff];
+
+    float[] initialStep = new float[numCoeff];
+    Arrays.fill(initialStep, 1.0f);
+    return optimize(1.0 / 512.0, iterations, initial, initialStep,
+        new ImmutablePair<Float, Float>(-10.0f, 10.0f), data);
+  }
+
+  public static float evaluatePolynomial(float[] coeff, float[] data, boolean overwrite) {
```

Comment:
why don't you also delete the existing implementation of these inside SplineTrainer?